torch/csrc/distributed/rpc/testing/faulty_process_group_agent.cpp

```diff
@@ -76,13 +81,45 @@ std::shared_ptr<FutureMessage> FaultyProcessGroupAgent::send(
   }
 }
 
+void FaultyProcessGroupAgent::enqueueSend(SendWork work) {
+  float msgDelay = getDelayForMessage(work.message_.type());
+  if (msgDelay != 0) {
+    // Sleep for the specified delay for the message.
+    std::this_thread::sleep_for(
+        std::chrono::milliseconds(static_cast<int>(msgDelay * 1000)));
```

Comment:
We should make this 1000 some constant like `kConvertToMilliseconds` (there's probably a better name to give this)